### PR TITLE
ref(reprocessing-v2): Expose group status and add more data to activity

### PR DESCRIPTION
**Pull request by untitaker**
* Expose "reprocessing" group status in group serializer and also add progress of reprocessing as group status details. This appears to not break the UI at all (I did thorough testing) but we're kind of treading on thin ice.

  @priscilawebdev To test this new status when developing UI, I recommend doing in sentry shell:

  ```python
  from sentry.reprocessing2 import start_group_reprocessing
  start_group_reprocessing(project_id=1, group_id=123)
  ```

  This will start reprocessing but actually never finish it. You can then no longer find the group in `is:unresolved`, you need to clear the search to be able to see it. You can also delete the group safely.

* Add more data about reprocessed event count to activity feed, + old/new group IDs to be able to link to search.
* Add documentation to reprocessing module
* bump ttl of redis counter such that we will have an easier time debugging hanging reprocessing jobs if it comes to that

### CODE REVIEW

File: src/sentry/reprocessing2.py [hunk 1]
```diff
@@ -1,3 +1,81 @@
...
+  This set is now appended to every query: `and group_id not in (<long list of
+  deleted group IDs>)`
+
+* Every n hours, ClickHouse folds rows with duplicate primary keys into one
+  row. Now only the `deleted=1` row of the deleted event remains. This process
+  is basically rewriting the table, and as such itself takes a couple of hours.
+
+  After that the Redis set can be cleared out.
+
+As such, reusing the group model will not work as the Redis set will prevent
```
Comment (untitaker) [inline src/sentry/reprocessing2.py:59]:
```suggestion
As such, reusing the group ID will not work as the Redis set will prevent
```

File: src/sentry/tasks/reprocessing2.py [hunk 2]
```diff
@@ -119,31 +133,52 @@ def reprocess_event(project_id, event_id, start_time):
...
-    name="sentry.tasks.reprocessing2.delete_old_group",
+    name="sentry.tasks.reprocessing2.finish_reprocessing",
     queue="events.reprocessing.preprocess_event",
     time_limit=(60 * 5) + 5,
     soft_time_limit=60 * 5,
 )
-def delete_old_group(project_id, group_id):
-    from sentry import similarity
-    from sentry.models.group import Group
+def finish_reprocessing(project_id, group_id, new_group_id):
```
Comment (jan-auer) [inline src/sentry/tasks/reprocessing2.py:156]: You should technically keep the old task around, but given that there is basically no usage on this at the moment it should be safe to remove the old task right away.

**Review by jan-auer — APPROVED** [review]
<3 for the docs.

File: src/sentry/tasks/reprocessing2.py (hunk 2 above)
Comment (untitaker) [reply src/sentry/tasks/reprocessing2.py:156]: i'm definetly yoloing that part especially with the changed function signatures across the board

**Pull request merged**